src/uncrustify.cpp

```diff
@@ -300,7 +300,7 @@ static void version_exit(void)
 }
 
 
-static void redir_stdout(const char *output_file)
+static int redir_stdout(const char *output_file)
```

Comment:
so easy to forget that this function returns an exit code. Maybe it should be part of it's interface?

Reply:
I don't follow. How would you make this "part of its interface" besides the manner in which it already is? Are you thinking:

```c++
int result;
redir_stdout(..., result);
if(result != EXIT_SUCCESS)
  exit(result);
```

Reply:
yeah, exactly. Even if I'm not 100% on board with this approach cause it "pollutes" it at the same time. Error handling is hard... Let's leave it as you did it.

Reply:
I am... not in favor of output parameters; a sentiment that seems to be widely shared among C++ programmers. That said, what did you think about my suggested `class ExitCode`? (Okay, besides that it should be `exit_code` :stuck_out_tongue:.) That might also document the intent more clearly, and I could additionally stick `[[nodiscard]]` on it... Having thought of it, I now rather like the idea, unless you think it would be too much clutter?

Reply:
I agree. I have not found a better way to do it other than returning a status value. Maybe we can make it an enum or a class as you say but I guess we'll over-engineer it too much.

Reply:
I think I've addressed all the comments; thanks for the review!